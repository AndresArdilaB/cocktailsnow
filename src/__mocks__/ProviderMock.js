import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
