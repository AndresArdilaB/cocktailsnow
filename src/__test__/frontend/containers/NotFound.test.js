import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import NotFound from '../../../frontend/containers/NotFound';

describe('<NotFound />', () => {

  test('Component render', () => {
    const notFound = shallow(
      <ProviderMock>
        <NotFound />
      </ProviderMock>,
    );

    expect(notFound.length).toEqual(1);
  });

});

describe('NotFound Snapshot', () => {

  test('Testing the NotFound UI component', () => {
    const notFound = create(<NotFound />);
    expect(notFound.toJSON()).toMatchSnapshot();
  });
});
