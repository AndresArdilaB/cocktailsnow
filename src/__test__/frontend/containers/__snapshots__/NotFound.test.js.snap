// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Snapshot Testing the NotFound UI component 1`] = `
Array [
  <h1>
    No Encontrado
  </h1>,
  <h2>
    Regresa al Home
  </h2>,
]
`;
